test/sample_test.cpp

```diff
@@ -131,9 +131,9 @@ TEST_CASE("rules sampler") {
     span->FinishWithOptions(finish_options);
 
     auto& metrics = mwriter->traces[0][0]->metrics;
-    REQUIRE(metrics.find("_dd.rule_psr") == metrics.end());
-    REQUIRE(metrics.find("_dd.limit_psr") == metrics.end());
-    REQUIRE(metrics.find("_dd.agent_psr") != metrics.end());
+    REQUIRE(metrics.find("_dd.rule_psr") != metrics.end());
+    REQUIRE(metrics.find("_dd.limit_psr") != metrics.end());
+    REQUIRE(metrics.find("_dd.agent_psr") == metrics.end());
```

Comment:
Does this mean the rates sent by the agent are never used? Or is there a new test required to handle that case?

Reply:
I don't think there's any configuration now that will cause the priority sampler to be used.  There's always a matching rule.

New test for which case?

Reply:
Per our conversation yesterday, now I understand what you mean.  Have since pushed changes to correct the behavior. I'll update the PR description, too...